src/url-utils.ts

```diff
@@ -30,6 +30,19 @@ const sharedRelativeUrlProperties =
 export type UrlString = string;
 
 /**
+ * Returns a URL with the basename removed from the pathname.  Strips the
+ * search off of the URL as well, since it will not apply.
+ */
+export function debaseUrl(href: UrlString): UrlString {
+  const u = url.parse(href);
+  if (typeof u.pathname === 'string') {
+    u.pathname = path.dirname(u.pathname + '_').replace(/_$/, '') + '/';
```

Comment:
I'll add a comment.  The reason for this is `path.dirname('/a/b/') incorrectly returns `/a`.  By suffixing the string with `_` I ensure that `/a/b/` is treated as `/a/b/_` and returns `/a/b`.  Also, `/a/b` is treated as `/a/b_` and therefore returns `/a` correctly.

The removing of the underscore was redundant as there should be no circumstance it remains.  That was a bug!  Thanks for bringing it up.  I have another code point where I do remove the underscore for a similar trick but that is not a bug.  This was.  Fixed and commented!